plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt

```diff
@@ -403,23 +403,56 @@ class LiquidDataflowTests {
   }
 
   @Test
-  @Disabled
-  fun `nullability, 1`() {
+  fun `nullability, predicate with null, 1`() {
     """
       ${imports()}
-      fun nully1(x: Int?): Int? {
+      fun nully1a(x: Int?): Int? {
         val y = x?.let { 1 }
-        return y.post({ it > 0 }) { "greater than 0" }
+        return y.post({ 
+          if (x == null) { it == null } 
```

Comment:
We can potentially simplify this introducing something like `nullable(x) { it > 0 }` which performs 
```kotlin
if (x == null) { it == null } 
else { (it != null) && (it > 0) } 
```
Still most users will understand what it is now as acceptable IMO